src/pages/home-page/model/store.ts

```diff
@@ -0,0 +1,44 @@
+import { createEffect, createEvent, createStore, sample } from 'effector';
+
+export const usernameChanged = createEvent<string>();
+export const passwordChanged = createEvent<string>();
+
+export const $usernameStore = createStore('');
+export const $passwordStore = createStore('');
+
+$usernameStore.on(usernameChanged, (_, username) => username);
+$passwordStore.on(passwordChanged, (_, password) => password);
+
+export const formSubmitted = createEvent<void>();
+
+const handleSubmitFx = createEffect(
+  async ({ username, password }: { username: string; password: string }) => {
```

Comment:
this is a service layer method. in entity, you should have async function, f.e loginUser, or authenticateCurrentUser...

in your store.ts in your entity you're creating the effect

`import { loginUser } from './api'`
`export const loginUserFx = createEffect(loginUser)`

and via Public API (index.ts) in entity you're exporting the whole public model, that can be used by higher layers